src/main/java/com/aws/greengrass/lifecyclemanager/Lifecycle.java

```diff
@@ -519,9 +521,23 @@ private void handleCurrentStateNew(Optional<State> desiredState) throws Interrup
             greengrassService.serviceErrored(ee);
         } catch (TimeoutException te) {
             greengrassService.serviceErrored(ComponentStatusCode.INSTALL_TIMEOUT, "Timeout in install");
-        } finally {
-            stopBackingTask();
         }
+
+        asyncFinishAction.set((Object stateEvent) -> {
+            // if a state is reported
```

Comment:
why is this the right async finish action?

For `handleCurrentStateInstalledAsync`, it just uses `stopBackingTask()` which seems more correct to me.

Can you explain why all this code is here?

Reply:
We want to set installation phase as background process, so we make some changes in GenericExternalService. After getting the new State from it's return infomation , then we can do some judgement here. But you pointed out that no changes should be made in GenericExternalService, so the change here doesn't seem to work either.